src/box/txn_limbo.h

```diff
@@ -162,13 +162,20 @@ struct txn_limbo {
 	 */
 	struct latch promote_latch;
 	/**
-	 * Maximal LSN gathered quorum and either already confirmed in WAL, or
-	 * whose confirmation is in progress right now. Any attempt to confirm
-	 * something smaller than this value can be safely ignored. Moreover,
-	 * any attempt to rollback something starting from <= this LSN is
-	 * illegal.
+	 * Maximal LSN that gathered quorum and has already been persisted in
```

Comment:
This assumption is broken, when we're trying to write `PROMOTE/DEMOTE`. You'll have not yet persisted state in `confirmed_vclock`, as `req_prepare` will change it

Reply:
See https://github.com/tarantool/tarantool/pull/10626#discussion_r1788125077.